lib/date_time/date_time.c

```diff
@@ -85,6 +85,18 @@ static void date_time_notify_event(const struct date_time_evt *evt)
 	if (app_evt_handler != NULL) {
 		app_evt_handler(evt);
 	}
+
+	/* If periodic updates are requested in the configuration
```

Comment:
I'm not sure if `date_time_notify_event` is the correct placement of this block of code. I would rather have a function that is called with this block of code before `date_time_notify_event` is called. The scope of this function is only to notifiy the application with library specific notifications.

Reply:
Will this apply for the call in `date_time_set` as well?

Reply:
Which is the asynchronous alternative to the internal date time update work.

Reply:
True. We don't need to run this new if-statement when called from `date_time_set()`, although it shouldn't harm either as the condition evaluates to false. I'll make another function for this.

Reply:
This is now done.